cli/src/cmd/bind.rs

```diff
@@ -0,0 +1,181 @@
+use crate::cmd::Cmd;
+
+use clap::{Parser, ValueHint};
+use ethers::contract::MultiAbigen;
+use foundry_config::{
+    figment::{
+        self,
+        error::Kind::InvalidType,
+        value::{Dict, Map, Value},
+        Metadata, Profile, Provider,
+    },
+    impl_figment_convert, Config,
+};
+use serde::Serialize;
+use std::{
+    path::{ PathBuf},
+};
+use tokio::fs;
+
+impl_figment_convert!(BindArgs);
+
+#[derive(Debug, Clone, Parser, Serialize)]
+pub struct BindArgs {
+    #[clap(
+        help = "The project's root path. By default, this is the root directory of the current Git repository or the current working directory if it is not part of a Git repository",
+        long,
+        value_hint = ValueHint::DirPath
+    )]
+    #[serde(skip)]
+    pub root: Option<PathBuf>,
+
+    #[clap(
+        help = "Path to where the contract artifacts are stored",
+        long = "out",
+        short,
+        value_hint = ValueHint::DirPath
+    )]
+    #[serde(rename = "out", skip_serializing_if = "Option::is_none")]
+    pub out_path: Option<PathBuf>,
+
+    #[clap(
+        help = "A directory at which to write the generated rust crate",
+        long = "bindings-root",
+        value_hint = ValueHint::DirPath
+    )]
+    #[serde(skip)]
+    pub bindings_root: Option<PathBuf>,
```

Comment:
not sure if this would work but what if we make this the first field of the struct

so that `forge bind "my-crate/src/contracts"` would be enough to generate the bindings.
for example if you're in foundry dir and already has a rust crate `./my-crate`

Reply:
also, perhaps we'll have more codegen use cases in the future, should we put this behind a `generate` subcommand?

Reply:
Should all code generation be under the same command? What additional codegen do we expect to need?

I was thinking in the future (after we get around to adding wasm-bindgen to the generated contract bindings that we may want a rust-default, TS second binding and publishing process:
 
```
forge bind
forge bind typescript

forge publish 
forge publish npm
```

with `publish` running `bind` and then publishing the package to crates.io / npm

Reply:
I like `bind` personally, and we can add more args to it for langs as you say here, so I wouldn't really change it